src/ert/scheduler/job.py

```diff
@@ -156,13 +134,37 @@ async def __call__(
         for attempt in range(max_submit):
             await self._submit_and_run_once(sem)
 
-            if self.returncode.cancelled() or (
-                self.returncode.done() and self.returncode.result() == 0
-            ):
+            if self.returncode.cancelled():
                 break
-            elif attempt < max_submit - 1:
-                message = f"Realization: {self.iens} failed, resubmitting"
+
+            if self.returncode.result() == 0:
+                callback_status, status_msg = forward_model_ok(self.real.run_arg)
+                if self._callback_status_msg != "":
+                    self._callback_status_msg = status_msg
+                else:
+                    self._callback_status_msg += (
+                        f"\nstatus from done callback: {status_msg}"
+                    )
+
+                if callback_status == LoadStatus.LOAD_SUCCESSFUL:
+                    await self._send(State.COMPLETED)
+                else:
+                    assert callback_status in (
+                        LoadStatus.LOAD_FAILURE,
+                        LoadStatus.TIME_MAP_FAILURE,
+                    )
+                    await self._send(State.FAILED)
+                break
+            elif self.returncode.result() != 0 and attempt < max_submit - 1:
```

Comment:
Do we need `self.returncode.result() != 0` here? It is already checked in the if-statement above.

Reply:
Lets keep it, it makes it easier to read.

Reply:
right!